File: Data Structures & Algorithms/two-integer-sum/submission-0.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map <int , int > maps ; 
        for ( int i = 0 ; i<nums.size() ; i++ ) 
        {    int need = target - nums[i] ; 
            if(maps.count(need ))
            { 
                return  { maps[need] , i } ; 
            }
            maps[nums[i]] = i ; 
        }
        return {};
        
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        res = []
        def dfs(c,count,total):
            if total == target :
                res.append(count.copy())
                return
            if total>target or c>=len(nums):
                return 
            count.append(nums[c])
            dfs(c,count,total+nums[c])
            count.pop()
            dfs(c+1,count,total)
        dfs(0,[],0)
        return res